### ISSUE TITLE
Whe does my Training loss didn't go down when I train lora of flux?

### ISSUE THREAD

**Issue by vincezh2000** [XLabs-AI/x-flux#43]
<img width="1400" alt="image" src="https://github.com/user-attachments/assets/92aa5fce-faed-4713-a568-eba2a16e3a27">
my loss curve looks something like this.

model_name: "flux-dev"
data_config:
  train_batch_size: 1
  num_workers: 4
  img_size: 512
  img_dir: xxx
report_to: wandb
train_batch_size: 1
output_dir: lora/
max_train_steps: 10000
learning_rate: 1e-4
lr_scheduler: constant
lr_warmup_steps: 10
adam_beta1: 0.9
adam_beta2: 0.999
adam_weight_decay: 0.01
adam_epsilon: 1e-8
max_grad_norm: 1.0
logging_dir: log16
mixed_precision: "bf16"
checkpointing_steps: 2500
checkpoints_total_limit: 10
tracker_project_name: lora_test
resume_from_checkpoint: latest
gradient_accumulation_steps: 2
rank: 16

**Comment by HelloWarcraft** [comment]
Loss will go down as long as the step is large enough, just wait. Every step in one iteration in Gradient Descent, if your lr is not suitable the loss might go up and even become `loss NAN`.

By the way, I met an error running `train_flux_lora_deepspeed.py`, https://github.com/XLabs-AI/x-flux/issues/44. 
🥺Could you give me some suggestions?

**Comment by vincezh2000** [comment]
It is obvious not the reason of step I think. But i am trying tuning the other parms

**Comment by HelloWarcraft** [comment]
> It is obvious not the reason of step I think. But i am trying tuning the other parms

@vincezh2000 I see, the reason is that the script doesn't caculate average loss. Average loss goes down in the gradient desecent process, but current loss might change in a vast range. 

I also found my 10000-step training loss curve didn't go down, but after the base model loaded the lora, the generated images look very similar to the images in the trainset.

**Comment by vincezh2000** [comment]
> > It is obvious not the reason of step I think. But i am trying tuning the other parms
> 
> @vincezh2000 I see, the reason is that the script doesn't caculate average loss. Average loss goes down in the gradient desecent process, but current loss might change in a vast range.
> 
> I also found my 10000-step training loss curve didn't go down, but after the base model loaded the lora, the generated images look very similar to the images in the trainset.

Thank you for this. I will check the training scirpt.

**Comment by arrowonstr** [comment]
the loss of lora is something mystical
Sometimes a high loss model may generate good image but low loss cannot
I think it's not the fault of flux beacuse in sd model i met the issue too
So  I rarely care about the loss , maybe it doesn't matter

**Comment by vincezh2000** [comment]
Like what @HelloWarcraft said. The model loged the loss for every epoch not for ave_loss.
![image](https://github.com/user-attachments/assets/9ab82673-3093-42d7-a50a-0fb08770691d)
I think that's why the loss doesn't go down.
![image](https://github.com/user-attachments/assets/fc3ce230-dc14-4008-8a30-9ddfcbbe6b48)
They just cal the avg loss for distrubuted training.
I am adding more params and triaing via this, will show the curve after I have the result

**Comment by hpkim0512** [comment]
@vincezh2000 Do you have any progress?

**Comment by Gbone3176** [comment]
@vincezh2000 I ran into almost the same problem as you, so I have no way to determine the fine-tuning training status, which frustrates me a lot.
